autogpt_platform/backend/backend/server/integrations/router.py

```diff
@@ -416,3 +418,72 @@ def _get_provider_oauth_handler(
         client_secret=client_secret,
         redirect_uri=f"{frontend_base_url}/auth/integrations/oauth_callback",
     )
+
+
+@router.get("/ayrshare/sso_url")
+async def get_ayrshare_sso_url(
+    user_id: Annotated[str, Depends(get_user_id)],
+) -> dict[str, str]:
+    """
+    Generate an SSO URL for Ayrshare social media integration.
+
+    Returns:
+        dict: Contains the SSO URL for Ayrshare integration
+    """
+    # Generate JWT and get SSO URL
+    client = AyrshareClient()
+
+    # Get or create profile key
+    profile_key = creds_manager.store.get_ayrshare_profile_key(user_id)
+    if not profile_key:
+        logger.info(f"Creating new Ayrshare profile for user {user_id}")
+        # Create new profile if none exists
+        profile = client.create_profile(title=f"User {user_id}", messaging_active=True)
+        if isinstance(profile, PostError):
+            logger.error(
+                f"Error creating Ayrshare profile for user {user_id}: {profile}"
+            )
+            raise HTTPException(
+                status_code=500, detail="Failed to create Ayrshare profile"
+            )
+        profile_key = profile.profileKey
+        creds_manager.store.set_ayrshare_profile_key(user_id, profile_key)
+    else:
+        logger.info(f"Using existing Ayrshare profile for user {user_id}")
+
+    # Convert SecretStr to string if needed
+    profile_key_str = (
+        profile_key.get_secret_value()
+        if isinstance(profile_key, SecretStr)
+        else str(profile_key)
```

Comment:
what type is `profile_key` in this case?

Reply:
a unique key aryshare generates for a user. In aryshare docs its called profile_key